cli/src/main/kotlin/tools/samt/cli/CliCompiler.kt

```diff
@@ -14,10 +15,14 @@ fun parseSourceFile(source: SourceFile, context: DiagnosticContext): FileNode? {
         return null
     }
 
-    return try {
+    val fileNode = try {
         Parser.parse(source, tokenStream, context)
     } catch (e: DiagnosticException) {
         // error message is added to the diagnostic console, so it can be ignored here
-        null
+        return null
     }
+
+    SemanticChecker.check(fileNode, context)
```

Comment:
I saw that these checks only take a single file into consideration, which is why its okay to perform them in here, but would it make sense to move them into the SemanticModelBuilder as well? What's your opinion on that?

Reply:
Yes they will be migrated later, the expression checks are next to follow and after that the annotations would follow, but that will likely follow in a separate PR

Reply:
Most checks have been moved into the semantic model builder, I left the unit-tests and implementation within the existing checker for the ones that weren't moved yet